README.md

```diff
@@ -1,207 +1,607 @@
+
+[raster-data-types]: https://github.com/stac-extensions/raster?tab=readme-ov-file#data-types
+[stac-1.1-data-types]: https://github.com/radiantearth/stac-spec/blob/bands/item-spec/common-metadata.md#data-types
+
+#### Input Structure Object
+
+| Field Name | Type                                   | Description                                                                                                                                                                                                               |
+|------------|----------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| shape      | \[integer]                             | **REQUIRED** Shape of the input n-dimensional array (e.g.: $B \times C \times H \times W$), including the batch size dimension. Each dimension must either be greater than 0 or -1 to indicate a variable dimension size. |
+| dim_order  | \[[Dimension Order](#dimension-order)] | **REQUIRED** Order of the `shape` dimensions by name.                                                                                                                                                                     |
+| data_type  | [Data Type Enum](#data-type-enum)      | **REQUIRED** The data type of values in the n-dimensional array. For model inputs, this should be the data type of the processed input supplied to the model inference function, not the data type of the source bands.   |
+
+A common use of `-1` for one dimension of `shape` is to indicate a variable batch-size.
+However, this value is not strictly reserved for the `b` dimension.
+For example, if the model is capable of automatically adjusting its input layer to adapt to the provided input data,
+then the corresponding dimensions that can be adapted can employ `-1` as well.
+
+#### Dimension Order
+
+Recommended values should use common names as much as possible to allow better interpretation by users and scripts
+that could need to resolve the dimension ordering for reshaping requirements according to the ML framework employed.
+
+Below are some notable common names recommended for use, but others can be employed as needed.
+
+- `batch`
+- `channel`
+- `time`
+- `height`
+- `width`
+- `depth`
+- `token`
+- `class`
+- `score`
+- `confidence`
+
+For example, a tensor of multiple RBG images represented as $B \times C \times H \times W$ should 
+indicate `dim_order = ["batch", "channel", "height", "width"]`.
+
+#### Normalize Enum
+
+Select one option from:
+- `min-max`
+- `z-score`
+- `l1`
+- `l2`
+- `l2sqr`
+- `hamming`
+- `hamming2`
+- `type-mask`
+- `relative`
+- `inf`
+- `clip`
+
+See [OpenCV - Interpolation Flags](https://docs.opencv.org/4.x/da/d54/group__imgproc__transform.html#ga5bb5a1fea74ea38e1a5445ca803ff121)
```

Comment:
I think this reference to interpolation flags needs to be switched with the later reference to normalization flags.

Reply:
Oh right! I definitely went up and down that document too often 😓